test/tests/api/worker/rest/EntityRestCacheTest.ts

```diff
@@ -1516,6 +1528,34 @@ export function testEntityRestCache(name: string, getStorage: (userId: Id) => Pr
 			}
 			unmockAttribute(mock)
 		})
+
+		o("loadMultiple should load necessary elements from the server, and get the rest from the cache if MailDetails is set", async function () {
+			const listId = "listId"
+
+			// no mailDetails and no mailDetailsDraft == LegacyMail
+			const legacyMail = createMailInstance(listId, "1", "1")
+			legacyMail.mailDetails = null
+			legacyMail.mailDetailsDraft = null
+			const inCache = [legacyMail, createMailInstance(listId, "3", "3")]
+
+			const mailDetailsMail = createMailInstance(listId, "1", "1")
+			const serverMails = [mailDetailsMail, createMailInstance(listId, "5", "5")]
+
+			await Promise.all(inCache.map(async (i) => await storage.put(i)))
+			const ids = inCache.concat(serverMails).map(getElementId)
+
+			const loadMultiple = spy((...any) => Promise.resolve(serverMails))
```

Comment:
I think the problem is that we're not testing the correct ids being passed to this, the mock server just returns all serverMails every time.